Objects/exceptions.c

```diff
@@ -1864,6 +1864,56 @@ ImportError_reduce(PyObject *self, PyObject *Py_UNUSED(ignored))
     return res;
 }
 
+static PyObject *
+ImportError_repr(PyObject *self)
+{
+    int hasargs = PyTuple_GET_SIZE(((PyBaseExceptionObject *)self)->args) != 0;
+    PyImportErrorObject *exc = PyImportErrorObject_CAST(self);
+    PyUnicodeWriter *writer = PyUnicodeWriter_Create(0);
+    if (writer == NULL) {
+        goto error;
+    }
+    PyObject *r = BaseException_repr(self);
+    if (PyUnicodeWriter_WriteSubstring(
+        writer, r, 0, PyUnicode_GET_LENGTH(r) - 1) < 0)
+    {
+        Py_XDECREF(r);
+        goto error;
+    }
+    Py_XDECREF(r);
```

Comment:
```suggestion
    PyObject *r = BaseException_repr(self);
    if (r == NULL) {
        goto error;
    }
    Py_ssize_t rlen = PyUnicode_GET_LENGTH(r);
    int rc = PyUnicodeWriter_WriteSubstring(writer, r, 0, rlen - 1);
    Py_DECREF(r);
    if (rc < 0) {
        goto error;
    }
```

The call to `repr` may fail so you need to handle it separately.

Reply:
ah well spotted!